### ISSUE TITLE
Add formatter

### ISSUE THREAD

**Issue by mark-wiemer** [luanti-org/docs.luanti.org#84]
Greatly reduces decision fatigue while writing documents :)

- CI fails if any file is not formatted
- Single well-documented command can be used by contributors to format the entire repo
- Files can easily be ignored or unignored

**Comment by wsor4035** [comment]
ci shouldnt be added till https://github.com/minetest/dev.luanti.org/issues/15 is done

**Comment by mark-wiemer** [comment]
Fair enough

**Comment by appgurueu** [comment]
Side note, it's also possible to have actions that directly commit to the working branches of PRs. I feel this could be neat to make the process smoother.

**Comment by mark-wiemer** [comment]
Yes, as long as the auto-committed code is thoroughly reviewed! Of course, that should apply to human-written code, but I think we've all had our oopsies with that. But yeah, I'm in favor of experimenting with this

**Comment by appgurueu** [comment]
A correct formatter should not require review of its output, it should only make changes that don't change anything semantically (like converting `-` to `*` in Markdown lists for example).

**Comment by mark-wiemer** [comment]
In my book, any changes to any files require review--you just never know!! But yes, formatters like Prettier should be fine, but we should always review as we're responsible for ensuring the stuff we merge and deploy is good, we can never "blame the formatter"

**Comment by grorp** [comment]
Prettier is nice

> ci shouldnt be added till https://github.com/minetest/dev.luanti.org/issues/15 is done

Why not? Wouldn't having automatic formatting make importing pages easier, since you'd have one thing less to worry about?

**Comment by wsor4035** [comment]
@mark-wiemer is there anything blocking cspell from being a ci action now and closing this issue?

**Comment by appgurueu** [comment]
Spell checking != formatting?

Let's just go with prettier.

As I've said, I think auto-formatting PRs would be nicer, but it's fine either way: If we only have the checks, contributors will just have to configure their editor appropriately for it to not be annoying.

**Comment by wsor4035** [comment]
i meant to comment on https://github.com/luanti-org/dev.luanti.org/issues/83

**Comment by mark-wiemer** [comment]
We can use prettier to both check and auto-format, which is what I'd prefer. But it's a bit complex to setup a "format and commit" step within a CI pipeline, so instead I usually have a "check formatting and error if issues" step in CI and a simple "apply formatting fixes" script for contributors to run locally

**Comment by wsor4035** [comment]
formatting was applied in https://github.com/luanti-org/docs.luanti.org/pull/201
